### PR TITLE
Hearing Sensor, the enemy can hear a sound and go to it

### PR DESCRIPTION
**Pull request by smbss1**

### CODE REVIEW

File: scripts/AI/Sensors/HearingSensor.gd [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class_name HearingSensor
+
+var eye_location: Vector2
+@onready var _enemy: Enemy = $".."
+
+signal sound_heared(source: Node2D, location: Vector2, intensity: float)
+
+func _ready():
+	if !GlobalSignals.sound_emitted.is_connected(_on_heard_sound):
+		GlobalSignals.sound_emitted.connect(_on_heard_sound)
```
Comment (edouard-sn) [inline scripts/AI/Sensors/HearingSensor.gd:11]: You can connect it in the GUI (right panel) and remove this part

image

File: scripts/AI/Sensors/HearingSensor.gd [hunk 2]
```diff
@@ -0,0 +1,19 @@
+extends Node2D
+class_name HearingSensor
+
+var eye_location: Vector2
+@onready var _enemy: Enemy = $".."
+
+signal sound_heared(source: Node2D, location: Vector2, intensity: float)
```
Comment (edouard-sn) [inline scripts/AI/Sensors/HearingSensor.gd:7]: `sound_heard`

File: scripts/Player.gd [hunk 3]
```diff
@@ -5,6 +5,7 @@ class_name Player
 @export var weapon_list : Dictionary
 
 var move_direction : Vector2 = Vector2.ZERO
+@onready var shoot_vfx: CPUParticles2D = $ShootVFX
```
Comment (edouard-sn) [inline scripts/Player.gd:8]: Is it shooting particle or sound particle ?

File: scripts/AI/States/Patrol.gd [hunk 4]
```diff
@@ -33,6 +33,8 @@ func enter(_msg := {}) -> void:
 		vision_sensor.can_see_target.connect(_on_see_target)
 	if !state_machine.navigation_agent.target_reached.is_connected(_on_target_reached):
 		state_machine.navigation_agent.target_reached.connect(_on_target_reached)
+	if !hearing_sensor.sound_heared.is_connected(_on_sound_heared):
```
Comment (edouard-sn) [inline scripts/AI/States/Patrol.gd:36]: Same here for the GUI stuff

**Review by edouard-sn — CHANGES_REQUESTED** [review]

**Comment by edouard-sn** [comment]
Btw idk if you did it but don't forget to merge main into this branch and see if everything works before merging

File: scripts/Player.gd (hunk 3 above)
Comment (smbss1) [reply scripts/Player.gd:8]: is "emitting sound" particle.
maybe something like `sound_shoot_vfx` name is better ?

File: scripts/AI/Sensors/HearingSensor.gd (hunk 1 above)
Comment (smbss1) [reply scripts/AI/Sensors/HearingSensor.gd:11]: I don't find how to connect the GlobalSignal from the GUI.
Do you know how to do it ?

File: scripts/Player.gd (hunk 3 above)
Comment (edouard-sn) [reply scripts/Player.gd:8]: Yeah I think so

File: scripts/AI/States/Patrol.gd (hunk 4 above)
Comment (smbss1) [reply scripts/AI/States/Patrol.gd:36]: I need to connect dynamicallly here but I saw that I forget to disconnect to the signal when quitting the state

**Pull request merged**